### PR TITLE
Feature/callback models bunq/sdk java#33

### LINKED ISSUE #33: Callback models for holding callback data
## Steps to reproduce:
1. Receive a callback as described here: https://doc.bunq.com/api/1/page/callbacks

## What should happen:
1. There is a model available so that you can do the following:
```php
$callbackModel = NotificationUrl::fromJson($CallbackJson);
$callbackModel->getReferencedObject();

if (!$callbackModel isntanceof CallbackModelImExcpeting) {
    throw new Exception();
}
```

## What happens:
1. Such thing is not possible yet

## Logs
- Logs `no logs`

## Extra info:
- Tested on [0.12.2](https://github.com/bunq/sdk_java/releases/tag/0.12.2)

## Reference
- https://github.com/bunq/sdk_php/issues/67

### LINKED ISSUE #37: ScheduledPayment causes decode error due to Typo
There is a typo in the `bunq\Model\Generated\Endpoint\SchedulePayment` class. It has the constant `const OBJECT_TYPE = 'SchedulePayment';` while the API returns an object of type `ScheduledPayment`. When I change the constant to this, it works fine.

## Steps to reproduce:
1. Run this piece of code: https://gist.github.com/mbernson/dd8c966e277b4acd820463335cd47e24

## What should happen:
1. I see a var_dump of my scheduled payments.

## What happens:
1. `PHP Notice:  Undefined index: SchedulePayment in /Users/mathijs/Code/bunqapibouwsel/vendor/bunq/sdk_php/src/Model/Core/BunqModel.php on line 115`

## Extra info:
- Tested on [bunq/sdk_java version 0.12.2](https://github.com/bunq/sdk_java/releases/tag/0.12.2) and the production bunq API

## Reference
- bunq/sdk_php#70

### PR DESCRIPTION
**Pull request by OGKevin**
Closes #33 
Closes #37

### CODE REVIEW

File: src/test/java/com/bunq/sdk/model/generated/object/NotificationUrlTest.java [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+        executeNotificationUrlTest(
+                JSON_PATH_MUTATION_MODEL,
+                Payment.class.getName(),
+                GETTER_PAYMENT
+        );
+    }
+
+    @Test
+    public void bunqMeTabModelTest() throws InvocationTargetException, FileNotFoundException,
+            ClassNotFoundException,
```
Comment (dnl-blkv) [inline src/test/java/com/bunq/sdk/model/generated/object/NotificationUrlTest.java]: Our unit of indentation in Java is _two spaces_, and two more for hanging code. It seems the indents are wider here %)

Please take a look at the rest of the SDK code and format consistently with it :)

File: src/test/java/com/bunq/sdk/model/generated/object/NotificationUrlTest.java [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+            "/ShareInviteBankInquiry.json";
+    private static final String JSON_PATH_SHARE_INVITE_BANK_RESPONSE_MODEL = BASE_PATH_JSON_MODEL +
+            "/ShareInviteBankResponse.json";
+
+    private static final Type NOTIFICATION_URL_TYPE = new TypeToken<NotificationUrl>(){}.getType();
+
+    private void executeNotificationUrlTest(
+            String expectedJsonFileName,
+            String classNameExpected,
+            String referencedObjectGetterName) throws FileNotFoundException, NoSuchMethodException,
```
Comment (dnl-blkv) [inline src/test/java/com/bunq/sdk/model/generated/object/NotificationUrlTest.java]: Multi-line argument formatting should have closing braces on new lines:
```
 private void executeNotificationUrlTest(
   String expectedJsonFileName,
   String classNameExpected,
   String referencedObjectGetterName
) throws FileNotFoundException, NoSuchMethodException,
```

Please fix this throughout the changes as well %)

File: src/test/java/com/bunq/sdk/model/generated/object/NotificationUrlTest.java [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+    private static final String JSON_PATH_REQUEST_INQUIRY_MODEL = BASE_PATH_JSON_MODEL + "/RequestInquiry.json";
+    private static final String JSON_PATH_REQUEST_RESPONSE_MODEL = BASE_PATH_JSON_MODEL + "/RequestResponse.json";
+    private static final String JSON_PATH_SCHEDULE_PAYMENT_MODEL = BASE_PATH_JSON_MODEL + "/ScheduledPayment.json";
+    private static final String JSON_PATH_SCHEDULE_INSTANCE_MODEL = BASE_PATH_JSON_MODEL + "/ScheduledInstance.json";
+    private static final String JSON_PATH_SHARE_INVITE_BANK_INQUIRY_MODEL = BASE_PATH_JSON_MODEL +
+            "/ShareInviteBankInquiry.json";
+    private static final String JSON_PATH_SHARE_INVITE_BANK_RESPONSE_MODEL = BASE_PATH_JSON_MODEL +
+            "/ShareInviteBankResponse.json";
+
+    private static final Type NOTIFICATION_URL_TYPE = new TypeToken<NotificationUrl>(){}.getType();
```
Comment (dnl-blkv) [inline src/test/java/com/bunq/sdk/model/generated/object/NotificationUrlTest.java]: Let us call it `TYPE_NOTIFICATION_URL` (from general to specific)

File: src/test/Recourse/NotificationUrlJsons/ScheduledPayment.json [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        },
+        "description": "Test payment to NL59BUNQ2025104669",
+        "merchant_reference": null,
+        "type": "BUNQ",
+        "sub_type": "PAYMENT"
+      }
+    }
+  }
+}
+
```
Comment (dnl-blkv) [inline src/test/Recourse/NotificationUrlJsons/ScheduledPayment.json]: Some of the JSON files in java seem to have _too many_ newlines in the end. It does not matter that much, but would be cool if you could fix it for consistency anyways :)

File: src/test/java/com/bunq/sdk/model/generated/object/NotificationUrlTest.java (hunk 1 above)
Comment (OGKevin) [reply src/test/java/com/bunq/sdk/model/generated/object/NotificationUrlTest.java]: My indention settings was on 4 😊 Changed to 2 and reformatted code.

File: src/test/java/com/bunq/sdk/model/generated/object/NotificationUrlTest.java [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Type;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+public class NotificationUrlTest extends BunqSdkTestBase {
+  /**
```
Comment (dnl-blkv) [inline src/test/java/com/bunq/sdk/model/generated/object/NotificationUrlTest.java]: Please add a newline before the class body beginning

File: src/test/Recourse/NotificationUrlJsons/ShareInviteBankResponse.json [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+          "view_new_events": true,
+          "budget": null,
+          "make_payments": true
+        }
+      },
+      "start_date": "2017-07-20 02:34:05.864029",
+      "end_date": "2017-11-11 02:34:05.864029"
+    }
+  }
+}
```
Comment (dnl-blkv) [inline src/test/Recourse/NotificationUrlJsons/ShareInviteBankResponse.json]: Json (all of it) should still be indented with 4 spaces xD

File: src/test/java/com/bunq/sdk/model/generated/object/NotificationUrlTest.java [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+  private static final String JSON_PATH_SCHEDULE_INSTANCE_MODEL = BASE_PATH_JSON_MODEL + "/ScheduledInstance.json";
+  private static final String JSON_PATH_SHARE_INVITE_BANK_INQUIRY_MODEL = BASE_PATH_JSON_MODEL +
+          "/ShareInviteBankInquiry.json";
+  private static final String JSON_PATH_SHARE_INVITE_BANK_RESPONSE_MODEL = BASE_PATH_JSON_MODEL +
+          "/ShareInviteBankResponse.json";
+
+  private static final Type TYPE_NOTIFICATION_URL = new TypeToken<NotificationUrl>() {}.getType();
+
+  private void executeNotificationUrlTest(
+          String expectedJsonFileName,
```
Comment (dnl-blkv) [inline src/test/java/com/bunq/sdk/model/generated/object/NotificationUrlTest.java]: Please take care of the hanging indent too. It should be 4 in our case I believe.

**Pull request merged**